### PR TITLE
Feature: Add Deletion Action to Manage Boxes

### PR DESCRIPTION
**Pull request by vahidbazzaz**
This PR introduces a delete option for managing boxes, along with a confirmation dialog for added security. Test cases 4.8.6 are fully implemented.

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 FE: add deletion to Manage Boxes v2](https://trello.com/c/2cEdEh2P/1524-20-fe-add-deletion-to-manage-boxes-v2)

File: front/src/hooks/useDeleteBoxes.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+          if (deletedBoxes.length) {
+            if (showToasts) {
+              createToast({
+                message: `${deletedBoxes.length === 1 ? "A box was" : `${deletedBoxes.length} boxes were`} successfully deleted.`,
+              });
+            }
+          }
+
+          if (invalidIdentifiers.length) {
```
Comment (pylipp) [inline front/src/hooks/useDeleteBoxes.ts:98]: @vahidbazzaz  Any boxes that are non-existing, already deleted, and/or in a base that the user must not access are returned in the `BoxResult.invalidBoxLabelIdentifiers` list.

File: front/src/hooks/useDeleteBoxes.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          labelIdentifiers,
+        },
+      })
+        .then(({ data, errors }) => {
+          setIsLoading(false);
+
+          if (errors?.length) {
+            const errorCode = errors[0]?.extensions?.code;
+
+            if (errorCode === "FORBIDDEN") {
```
Comment (pylipp) [inline front/src/hooks/useDeleteBoxes.ts:62]: For permission errors, the mutations won't return FORBIDDEN error code but an InsufficientPermissionError as part of the DeleteBoxesResult union.

File: front/src/types/generated/graphql.ts [hunk 3]
```diff
@@ -3,17 +3,15 @@ export type InputMaybe<T> = Maybe<T>;
 export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
 export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
 export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
-export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
-export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
 /** All built-in and custom scalars, mapped to their actual values */
 export type Scalars = {
-  ID: { input: string; output: string; }
```
Comment (pylipp) [inline front/src/types/generated/graphql.ts:10]: Please update graphql-codegen via `pnpm update` and re-run :)

**Comment by HaGuesto** [comment]
Also @vahidbazzaz would be great if you could add some tests. Then you would have noticed that the error handling would fail atm.

**Review by HaGuesto — APPROVED** [review]
@vahidbazzaz great work.

I gave it a few final touches. I hope you do not mind:
- the hook is now creating all toast. 
- no refetch is needed anymore.
- center the delete icon in the button on mobile screens

**Pull request merged**